webdriver/tests/bidi/network/fail_request/invalid.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+import webdriver.bidi.error as error
+
+pytestmark = pytest.mark.asyncio
+
+PAGE_EMPTY_TEXT = "/webdriver/tests/bidi/network/support/empty.txt"
+
+
+@pytest.mark.parametrize("value", [None, False, 42, {}, []])
+async def test_params_request_invalid_type(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.fail_request(request=value)
+
+
+@pytest.mark.parametrize("value", ["", "foo"])
+async def test_params_request_invalid_value(bidi_session, value):
+    with pytest.raises(error.NoSuchRequestException):
+        await bidi_session.network.fail_request(request=value)
+
+
+async def test_params_request_no_such_request(bidi_session, setup_network_test,
+                                              wait_for_event, fetch, url):
+    network_events = await setup_network_test(events=[
+        "network.responseCompleted",
+    ])
+
+    text_url = url(PAGE_EMPTY_TEXT)
+    on_response_completed = wait_for_event("network.responseCompleted")
+    await fetch(text_url)
+    await on_response_completed
+
+    request = network_events["network.responseCompleted"][0]["request"]["request"]
```

Comment:
```suggestion
    response_completed_event = await on_response_completed

    request = response_completed_event["request"]["request"]
```

We could simplify a bit and avoid reading the array.

Reply:
Done